### PR TITLE
Fix #135: Remove volatile/non volatile compile warning.

### LINKED ISSUE #135: Volatile/non volatile errors in CDC module
The version of STM32 HAL USB code we're using generates a number of warnings during compiles that should be corrected (maybe update to current first).

```console
C:\Users\tmm\AppData\Local\arduino15\packages\mcci\hardware\stm32\2.6.0\cores\arduino\stm32\USB\usbd_cdc_if.c: In function 'CDC_Receive_FS':
C:\Users\tmm\AppData\Local\arduino15\packages\mcci\hardware\stm32\2.6.0\cores\arduino\stm32\USB\usbd_cdc_if.c:279:14: warning: passing argument 1 of 'memcpy' discards 'volatile' qualifier from pointer target type [-Wdiscarded-qualifiers]
       memcpy(&UserRxBufferFS[UserRxBufPtrIn], Buf, n);
              ^
C:\Users\tmm\AppData\Local\arduino15\packages\mcci\hardware\stm32\2.6.0\cores\arduino\stm32\USB\usbd_cdc_if.c:281:16: warning: passing argument 1 of 'memcpy' discards 'volatile' qualifier from pointer target type [-Wdiscarded-qualifiers]
         memcpy(UserRxBufferFS, &Buf[n], Size);
                ^~~~~~~~~~~~~~
C:\Users\tmm\AppData\Local\arduino15\packages\mcci\hardware\stm32\2.6.0\cores\arduino\stm32\USB\usbd_cdc_if.c: In function 'CDC_TransmitDone':
C:\Users\tmm\AppData\Local\arduino15\packages\mcci\hardware\stm32\2.6.0\cores\arduino\stm32\USB\usbd_cdc_if.c:306:45: warning: passing argument 2 of 'USBD_CDC_SetTxBuffer' discards 'volatile' qualifier from pointer target type [-Wdiscarded-qualifiers]
     USBD_CDC_SetTxBuffer(&hUSBD_Device_CDC, gCDC_pTxBufferFS, UserTxBufPtrIn);
                                             ^~~~~~~~~~~~~~~~
C:\Users\tmm\AppData\Local\arduino15\packages\mcci\hardware\stm32\2.6.0\cores\arduino\stm32\USB\usbd_cdc_if.c: In function 'CDC_Write':
C:\Users\tmm\AppData\Local\arduino15\packages\mcci\hardware\stm32\2.6.0\cores\arduino\stm32\USB\usbd_cdc_if.c:442:12: warning: passing argument 1 of 'memcpy' discards 'volatile' qualifier from pointer target type [-Wdiscarded-qualifiers]
     memcpy(StackTxBufferFS, pBuffer, nBytes);
            ^~~~~~~~~~~~~~~
C:\Users\tmm\AppData\Local\arduino15\packages\mcci\hardware\stm32\2.6.0\cores\arduino\stm32\USB\usbd_cdc_if.c:456:12: warning: passing argument 1 of 'memcpy' discards 'volatile' qualifier from pointer target type [-Wdiscarded-qualifiers]
     memcpy(gCDC_pTxBufferFS + UserTxBufPtrIn, pBuffer, nBytes);
            ^~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by chwon64**

### CODE REVIEW

File: cores/arduino/stm32/USB/usbd_cdc_if.c [hunk 1]
```diff
@@ -276,14 +276,14 @@ static int8_t CDC_Receive_FS(uint8_t* Buf, uint32_t *Len)
     if ((UserRxBufPtrIn + Size) > APP_RX_DATA_SIZE)
     {
       uint32_t n = APP_RX_DATA_SIZE - UserRxBufPtrIn;
-      memcpy(&UserRxBufferFS[UserRxBufPtrIn], Buf, n);
+      memcpy((void *)&UserRxBufferFS[UserRxBufPtrIn], Buf, n);
```
Comment (terrillmoore) [inline cores/arduino/stm32/USB/usbd_cdc_if.c]: maybe add `/* not-volatile */` comment? i.e.

```c
memcpy((/* not-volatile */ void *)&UserRxBufferFS[UserRxBufPtrIn], Buf, n);
```

Comment (chwon64) [reply cores/arduino/stm32/USB/usbd_cdc_if.c]: Okay, I will add comment as suggested.
Regarding #78, I think current RX buffer size is configured as 256 bytes for both RX and TX.
{code}
#define APP_RX_DATA_SIZE  256
#define APP_TX_DATA_SIZE  256
{code}

Comment (chwon64) [reply cores/arduino/stm32/USB/usbd_cdc_if.c]: Do you want to change RX and TX buffer size for #78?

Comment (terrillmoore) [reply cores/arduino/stm32/USB/usbd_cdc_if.c]: #78 is a bug. Every so often (when it rolls around perhaps?) the entire previous contents of the RX buffer gets replayed as input. No need to change input size, but definite need to figure out why this is happening (and correct it). It's easy to duplicate; just create a long string of echo commands, and enter them one at a time. For example:

1. reset Catena

2. copy/paste these lines (around 254 characters):

    ```console
    echo 0123456789.........................................................................
    echo 9876543210..............................................................................
    echo abcdefg123....................................................
    ```

3. type a few more characters, and youi'll see that the echos show up again.

**Pull request merged**